manager/integration/tests/test_statefulset.py

```diff
@@ -0,0 +1,311 @@
+    """
+    for _, client in clients.iteritems():
+        break
+
+    statefulset_name = 'statefulset-backup-test'
+    statefulset['metadata']['name'] = \
+        statefulset['spec']['selector']['matchLabels']['app'] = \
+        statefulset['spec']['serviceName'] = \
+        statefulset['spec']['template']['metadata']['labels']['app'] = \
+        statefulset_name
+    (statefulset['spec']['volumeClaimTemplates'][0]['spec']
+        ['storageClassName']) = DEFAULT_STORAGECLASS_NAME
+    storage_class['metadata']['name'] = DEFAULT_STORAGECLASS_NAME
+
+    create_storage_class(storage_class)
+    create_and_wait_statefulset(statefulset)
+
+    volumes = []
+    for i in range(statefulset['spec']['replicas']):
+        pvc_name = (statefulset['spec']['volumeClaimTemplates'][0]['metadata']
+                    ['name'] + '-' + statefulset['metadata']['name'] + '-' +
+                    str(i))
+        volume_name = get_volume_name(core_api, pvc_name)
+        volumes.append({
+            'volume_name': volume_name,
+            'pod_name': statefulset['metadata']['name'] + '-' + str(i),
+            'data': generate_random_data(VOLUME_RWTEST_SIZE),
+            'backup_snapshot': ''
+        })
+
+    for v in volumes:
+        # Create backup.
+        volume = client.by_id_volume(v['volume_name'])
+        volume.snapshotCreate()
+        write_volume_data(core_api, v['pod_name'], v['data'])
+        v['backup_snapshot'] = volume.snapshotCreate()
+        volume.snapshotCreate()
+        volume.snapshotBackup(name=v['backup_snapshot']['name'])
+
+        # Wait for backup to appear.
+        found = False
+        for i in range(100):
+            backup_volumes = client.list_backupVolume()
+            for bv in backup_volumes:
+                if bv['name'] == v['volume_name']:
+                    found = True
+                    break
+            if found:
+                break
+            time.sleep(DEFAULT_POD_INTERVAL)
+        assert found
+
+        found = False
+        for i in range(20):
+            backups = bv.backupList()
+            for b in backups:
+                if b['snapshotName'] == v['backup_snapshot']['name']:
+                    found = True
+                    break
+            if found:
+                break
+            time.sleep(DEFAULT_POD_INTERVAL)
+        assert found
+
+        # Make sure backup has the correct attributes.
+        new_b = bv.backupGet(name=b["name"])
+        assert new_b["name"] == b["name"]
+        assert new_b["url"] == b["url"]
+        assert new_b["snapshotName"] == b["snapshotName"]
+        assert new_b["snapshotCreated"] == b["snapshotCreated"]
+        assert new_b["created"] == b["created"]
+        assert new_b["volumeName"] == b["volumeName"]
+        assert new_b["volumeSize"] == b["volumeSize"]
+        assert new_b["volumeCreated"] == b["volumeCreated"]
+
+
+def test_statefulset_recurring_backup(clients, core_api, storage_class,  # NOQA
```

Comment:
tag it with `@pytest.mark.recurring_job` so we can skip it by default. This will take a longer time than other cases.